### PR TITLE
Blockchain Updates: add transactions metadata, and recipient addresses to it

### PR DESCRIPTION
**Pull request by dvshur**

### CODE REVIEW

File: proto/waves/events/events.proto [hunk 1]
```diff
@@ -96,3 +99,22 @@ message StateUpdate {
         bytes safe_volume = 21;
     }
 }
+
+message TransactionMetadata {
+    bytes id = 1;
+
+    oneof metadata {
+        // <TransactionName>Metadata <transaction_name> = 1<TransactionNumber>
```
Comment (tsigel) [inline proto/waves/events/events.proto:107]: Добавить dApp для инвока (адрес)

Comment (tsigel) [reply proto/waves/events/events.proto:107]: Function name and arguments for invoke script.

Comment (tsigel) [reply proto/waves/events/events.proto:107]: State changes for invoke

File: proto/waves/events/events.proto [hunk 2]
```diff
@@ -96,3 +97,45 @@ message StateUpdate {
         bytes safe_volume = 21;
     }
 }
+
+message TransactionMetadata {
+    oneof metadata {
+        // <TransactionName>Metadata <transaction_name> = 1<TransactionNumber>
+        TransferMetadata transfer = 104;
+        MassTransferMetadata mass_transfer = 111;
+        InvokeScriptMetadata invoke_script = 116;
```
Comment (dvshur) [inline proto/waves/events/events.proto:106]: LeaseMetadata, needs recipient_address as well.

File: proto/waves/events/events.proto [hunk 3]
```diff
@@ -96,3 +97,45 @@ message StateUpdate {
...
+        string function_name = 2;
+        repeated Argument arguments = 3;
+        repeated Payment payments = 4;
+        
+        message Argument {
+            message List {
+                repeated Argument items = 1;
+            }
+
+            oneof value {
```
Comment (dvshur) [inline proto/waves/events/events.proto:129]: Todo: add all supported types

File: proto/waves/events/events.proto [hunk 4]
```diff
@@ -96,3 +97,45 @@ message StateUpdate {
...
+    
+    message MassTransferMetadata {
+        repeated bytes recipients_addresses = 1;
+    }
+
+    message InvokeScriptMetadata {
+        bytes d_app_address = 1;
+        string function_name = 2;
+        repeated Argument arguments = 3;
+        repeated Payment payments = 4;
```
Comment (dvshur) [inline proto/waves/events/events.proto:122]: Заменить `payments` на `transfers`

**Pull request closed without merging**